vector/src/main/java/im/vector/app/features/home/room/list/SectionHeaderAdapter.kt

```diff
@@ -44,11 +45,12 @@ class SectionHeaderAdapter constructor(
             val shouldShowExpandedArrow: Boolean = false
     )
 
-    lateinit var roomsSectionData: RoomsSectionData
+    var roomsSectionData: RoomsSectionData = roomsSectionData
```

Comment:
strange line of code 🤔

Reply:
ah it's using the value from the constructor. A bit confusing

Reply:
it is to prevent the lateinit declaration of the attribute and the initialization check of the attribute which was previously done in the `updateSection` method. I force passing the value as a constructor param and I init the attribute with that value. Previously, the attribute was initialized directly after creating the class object. I could set a different name to the param for clarity.